riscv/src/register/tests.rs

```diff
@@ -0,0 +1,25 @@
+read_write_csr! {
+    "test CSR register type",
+    Mtest: 0x000,
+    0b1111_1111,
+    "test single-bit field",
+    single: 0,
+    "setter test single-bit field",
+    set_single: 0,
+}
+
+read_write_csr_field! {
+    Mtest,
+    "multiple single-bit field range",
+    multi_range: 1..=3,
+    "setter multiple single-bit field range",
+    set_multi_range: 1..=3,
+}
+
+read_write_csr_field!(
+    Mtest,
+    "multi-bit field",
+    multi_field: [4:7],
+    "setter multi-bit field",
+    set_multi_field: [4:7],
+);
```

Comment:
Test calls to the new macros. The above is just a basic compile test.

Still need to add unit tests, and macro calls for the `read-only` and `write-only` variants.